### DISCUSSION TITLE
Import into a table from .sql file

### DISCUSSION THREAD

**Discussion by amaltus** [pawelsalawa/sqlitestudio#4966] [Q&A] [answered]
Hi. I've been using SQLiteStudio for several years now and now I need to import 30,000 records from a .sql file into a database. Using sqlite3 this takes several hours. In SQLiteStudio I only see the CSV option. Perhaps I'm missing something. How do I import data from a .sql file? Thanks.

**Comment by pawelsalawa** [comment, marked as answer]
Have you tried using "Execute SQL from file" option from a context menu? Right-click on a database and there is that option.

**Reply by pawelsalawa** [reply]
...although if for `sqlite3` it takes several hours, I would not expect SQLiteStudio to do it any faster.

**Comment by amaltus** [comment]
Thanks, this is just what I needed. I tested sqlite3 again and got execution in 3 hours. SQLiteStudio completed the job in 5 seconds. 
[18:05:37] Finished executing 23621 queries in 5.292 seconds.
Amazing! I wonder what's wrong with sqlite3.
